src/psyclone/tests/domain/lfric/kernel/field_vector_arg_metadata_test.py

```diff
@@ -0,0 +1,175 @@
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+# FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+# COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+# INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+# BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+# ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+# -----------------------------------------------------------------------------
+# Author R. W. Ford, STFC Daresbury Lab
+
+'''Module containing tests for the FieldVectorArgMetadata class.
+
+'''
+import pytest
+
+from fparser.two import Fortran2003
+
+from psyclone.domain.lfric.kernel import FieldVectorArgMetadata
+
+
+def test_create():
+    '''Test that an instance of FieldVectorArgMetadata can be created
+    successfully.
+
+    '''
+    field_vector_arg = FieldVectorArgMetadata("GH_REAL", "GH_READ", "W0", "2")
+    assert isinstance(field_vector_arg, FieldVectorArgMetadata)
+    assert field_vector_arg.form == "gh_field"
+    assert field_vector_arg._datatype == "gh_real"
+    assert field_vector_arg._access == "gh_read"
+    assert field_vector_arg._function_space == "w0"
+    assert field_vector_arg._vector_length == "2"
+
+
+def test_init_invalid():
+    '''Test that an invalid vector length supplied to the constructor
+    raises the expected exception.
+
+    '''
+    with pytest.raises(TypeError) as info:
+        _ = FieldVectorArgMetadata("GH_REAL", "GH_READ", "W0", 1)
+    assert ("The vector size should be a string but found int."
+            in str(info.value))
+
+
+def test_create_from_fparser2():
+    '''Test that the create_from_fparser2 static method works as
+    expected. Test that all relevant check and get methods are called
+    by raising exceptions within them, as well as checking for valid
+    input.
+
+    '''
+    with pytest.raises(TypeError) as info:
+        _ = FieldVectorArgMetadata.create_from_fparser2("hello")
+    assert ("Expected kernel metadata to be encoded as an fparser2 "
+            "Part_Ref object but found type 'str' with value 'hello'."
+            in str(info.value))
+
+    fparser2_tree = FieldVectorArgMetadata.create_fparser2(
```

Comment:
As I've commented in InterGridVectorArg, the fact that we have so much duplication between this test and the one for FieldArgMetadata doesn't seem right to me.

Reply:
Yes, agreed. I've hopefully addressed this now.